logger/src/commonMain/kotlin/com/wire/kalium/logger/KaliumLogger.kt

```diff
@@ -61,96 +62,143 @@ enum class KaliumLogLevel {
     DISABLED
 }
 
+fun KaliumLogLevel.toMinSeverity(): Severity = when (this) {
+    KaliumLogLevel.VERBOSE -> Severity.Verbose
+    KaliumLogLevel.DEBUG -> Severity.Debug
+    KaliumLogLevel.INFO -> Severity.Info
+    KaliumLogLevel.WARN -> Severity.Warn
+    KaliumLogLevel.ERROR -> Severity.Error
+    KaliumLogLevel.DISABLED -> Severity.Assert
+}
+
+fun Severity.toKaliumLogLevel(): KaliumLogLevel = when (this) {
+    Severity.Verbose -> KaliumLogLevel.VERBOSE
+    Severity.Debug -> KaliumLogLevel.DEBUG
+    Severity.Info -> KaliumLogLevel.INFO
+    Severity.Warn -> KaliumLogLevel.WARN
+    Severity.Error -> KaliumLogLevel.ERROR
+    Severity.Assert -> KaliumLogLevel.DISABLED
+}
+
 /**
  * the logWriter is to create a custom writer other than the existing log writers from kermit to intercept the logs
  * in the android case we use it to write the logs on file
  *
  */
-class KaliumLogger(private val config: Config, vararg logWriters: LogWriter = arrayOf()) {
-
-    private var kermitLogger: KermitLogger
-
-    init {
-        kermitLogger = if (logWriters.isEmpty()) {
-            KermitLogger(
-                config = StaticConfig(
-                    minSeverity = config.severityLevel, listOf(platformLogWriter())
-                ),
-                tag = config.tag
-            )
-        } else {
-            KermitLogger(
-                config = StaticConfig(
-                    minSeverity = config.severityLevel, logWriters.asList()
-                ),
-                tag = config.tag
-            )
-        }
+class KaliumLogger(
+    private val config: Config = Config.DEFAULT,
+    private val tag: Tag = Tag.Text("KaliumLogger")
+) {
+
+    constructor(
+        config: Config = Config.DEFAULT,
+        tag: String = "KaliumLogger"
+    ) : this(config, Tag.Text(tag))
+    private val kermitLogger: KermitLogger = KermitLogger(
+        config = config.kermitConfig()
+    )
+
+    private fun tag(): String = when (tag) {
+        is Tag.Text -> tag.text
+        is Tag.UserClientText -> tag.data().let { "${tag.prefix}[${it.userId}|${it.clientId}]" }
     }
 
-    val severity = config.severity
+    fun logLevel(): KaliumLogLevel = config.logLevel()
 
     @Suppress("unused")
-    fun withFeatureId(featureId: ApplicationFlow): KaliumLogger {
-        val currentLogger = this
-        currentLogger.kermitLogger = kermitLogger.withTag("featureId:${featureId.name.lowercase()}")
-        return currentLogger
-    }
+    fun withFeatureId(featureId: ApplicationFlow): KaliumLogger = KaliumLogger(
+        config = config,
+        tag = "featureId:${featureId.name.lowercase()}".let {
+            when (tag) {
+                is Tag.Text -> Tag.Text(it)
+                is Tag.UserClientText -> Tag.UserClientText(it, tag.data)
+            }
+        }
+    )
+
+    fun withUserDeviceData(data: () -> UserClientData): KaliumLogger = KaliumLogger(
```

Comment:
suggestion: also here. Document to make sure the caller understands the behaviour.